examples/text_to_image/train_text_to_image_sdxl.py

```diff
@@ -540,7 +542,7 @@ def compute_vae_encodings(batch, vae):
     with torch.no_grad():
         model_input = vae.encode(pixel_values).latent_dist.sample()
     model_input = model_input * vae.config.scaling_factor
-    return {"model_input": model_input.cpu()}
+    return {"model_input": accelerator.gather(model_input)}
```

Comment:
Sorry if my question is stupid, but why do we need to gather here? Doesn't this cause a sync between all ranks, as opposed to npu to cpu memory sync, making it slower overall?